plugins/org.obeonetwork.dsl.uml2.design/src/org/obeonetwork/dsl/uml2/design/internal/services/AssociationServices.java

```diff
@@ -51,6 +53,26 @@ public Property getSource(Association association) {
 	}
 
 	/**
+	 * Get the source of an n-ary association.
+	 * @param association the association
+	 * @param view the edge to retrieve the source end
+	 * @return end
+	 */
+	public Property getSourceEndAssociation(Association association, DDiagramElement view) {
+		final DEdge edge = (DEdge)view;
+		final String sourceObject = edge.getSourceNode().toString();
```

Comment:
Instead of using toString use the EObject to compare see below for an example : 
private Property getSourceEndNaryAssociation(Association association, DDiagramElement view){
		final DEdge edge = (DEdge)view;
		final List<Property> members = association.getMemberEnds();

		final DNodeListSpec source = (DNodeListSpec)edge.getSourceNode();
		final List<EObject> sourceSemanticElements = source.getSemanticElements();
		for (final EObject sourceEObject : sourceSemanticElements) {
			for (final Property member : members) {
				if (sourceEObject.equals(member.getType())) {
					return member;
				}
			}
		}

		return null;
	}